js/mkdir.ts

```diff
@@ -3,33 +3,48 @@ import * as msg from "gen/msg_generated";
 import * as flatbuffers from "./flatbuffers";
 import * as dispatch from "./dispatch";
 
-/** Creates a new directory with the specified path and permission
- * synchronously.
+/** Creates a new directory with the specified path synchronously.
+ * If `recursive` is set to true, nested directories will be created (also known
+ * as "mkdir -p").
+ * `mode` sets permission bits (before umask) on UNIX and does nothing on
+ * Windows.
  *
  *       import { mkdirSync } from "deno";
  *       mkdirSync("new_dir");
+ *       mkdirSync("nested/directories", true);
  */
-export function mkdirSync(path: string, mode = 0o777): void {
-  dispatch.sendSync(...req(path, mode));
+export function mkdirSync(path: string, recursive = false, mode = 0o777): void {
```

Comment:
This function should not take 3 arguments. Make the second argument an options object. So it effectively becomes:

```ts
interface MkdirOptions {
  recursive: boolean;
  mode: number;
}
`mkdirSync(path, { recursive = false, mode = 0o777 }: MkdirOptions = {})` {
...
```

Reply:
Done.